nmtwizard/preprocess/tokenizer.py

```diff
@@ -66,27 +67,36 @@ def make_subword_learner(subword_config, subword_dir, tokenizer=None):
 
 def vocabulary_iterator(vocabulary_path):
     """Iterates over each token included in the vocabulary file."""
-    with open(vocabulary_path) as vocabulary_file:
-        header = True
-        for line in vocabulary_file:
-            # The vocabulary file might start with some comments prefixed with '#'.
-            if header and line[0] == '#':
-                continue
-            header = False
-            line = line.rstrip('\n\r')
-            fields = line.split(' ')
-            if len(fields) == 1:
-                # No frequency value, the line is just the token.
-                yield fields[0]
-            else:
-                # The code below checks the last field is a frequency and not a part of
-                # a badly formatted token.
-                try:
-                    float(fields[-1])
-                    fields.pop()
-                except ValueError:
-                    pass
-                yield ' '.join(fields)
+    if not isinstance(vocabulary_path, set):
```

Comment:
Where is this function called on a set? Can we use instead the function `load_vocabulary` below that works for both a file and a set?

Reply:
The bug came from the call here : 

https://github.com/OpenNMT/nmt-wizard-docker/blob/3e7e0da30a151124bc73aaeecc08e1753d63b641/nmtwizard/preprocess/operators/tokenization.py#L30

Indeed 'load_vocabulary' should work, sorry I didn't see it.